### PR TITLE
bugfix: fix the issue of Raft registry exception without waiting for retries

### LINKED ISSUE #6048: When all servers in a raft cluster are offline, the client's watch requests will loop indefinitely.
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
当server在raft模式下全部宕机时，客户端将会无限循环尝试watch，但是最终都失败，因为内层的dopost方式将io异常catch了，导致无法sleep 1s
When the servers are all down in draft mode, the client will try to watch in an infinite loop, but eventually fail because the inner dopost method catches the io exceptions, making it impossible to sleep for 1s.

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

Minimal yet complete reproducer code (or URL to code):

### Ⅵ. Environment:

- JDK version(e.g. `java -version`):
- Seata client/server version:  
- Database version:
- OS(e.g. `uname -a`):
- Others:

### PR DESCRIPTION
**Pull request by funky-eyes**
- [ ] I have registered the PR [changes](../changes).

### Ⅱ. Does this pull request fix one issue?

fixes #6048

### CODE REVIEW

File: discovery/seata-discovery-raft/src/main/java/io/seata/discovery/registry/raft/RaftRegistryServiceImpl.java [hunk 1]
```diff
@@ -367,34 +367,28 @@ public static CloseableHttpResponse doGet(String url, Map<String, String> param,
...
-            httpPost.setHeader("Content-Type", "application/x-www-form-urlencoded");
-            client = HTTP_CLIENT_MAP.computeIfAbsent(timeout,
-                k -> HttpClients.custom().setConnectionManager(POOLING_HTTP_CLIENT_CONNECTION_MANAGER)
-                    .setDefaultRequestConfig(RequestConfig.custom().setConnectionRequestTimeout(timeout)
-                        .setSocketTimeout(timeout).setConnectTimeout(timeout).build())
-                    .build());
-            return client.execute(httpPost);
-        } catch (Exception e) {
-            LOGGER.error(e.getMessage(), e);
+        int timeout) throws Exception {
```
Comment (slievrly) [inline discovery/seata-discovery-raft/src/main/java/io/seata/discovery/registry/raft/RaftRegistryServiceImpl.java]: No business processing relevance, it is suggested to move this logic to httpUtil.

Comment (funky-eyes) [reply discovery/seata-discovery-raft/src/main/java/io/seata/discovery/registry/raft/RaftRegistryServiceImpl.java]: 目前没有通用的httputil,是要额外创建一个吗?
There is no generic httputil, do I have to create an extra one?

Comment (funky-eyes) [reply discovery/seata-discovery-raft/src/main/java/io/seata/discovery/registry/raft/RaftRegistryServiceImpl.java]: down

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**